core/src/main/scala/org/http4s/multipart/Multipart.scala

```diff
@@ -19,10 +19,26 @@ package multipart
 
 import org.http4s.headers._
 
-final case class Multipart[F[_]](parts: Vector[Part[F]], boundary: Boundary = Boundary.create) {
+/** Create a new multipart from a vector of parts and a boundary.
+  *
+  * Caution: the default argument for boundary is impure.  There is no
+  * binary-compatible fix, but the default parameter will be dropped
+  * in 1.0.
+  */
+final case class Multipart[F[_]](
+    parts: Vector[Part[F]],
+    boundary: Boundary,
+) {
   def headers: Headers =
     Headers(
       `Transfer-Encoding`(TransferCoding.chunked),
       `Content-Type`(MediaType.multipartType("form-data", Some(boundary.value))),
     )
 }
+
+object Multipart {
+  @deprecated("Retaining for binary-compatibility", "0.22.14")
+  def `<init>$default$2`: String = apply$default$2
+  @deprecated("Retaining for binary-compatibility", "0.22.14")
+  def apply$default$2: String = Boundary.unsafeCreate().value
```

Comment:
If we leave them public, then we don't have to fight mima.

Reply:
What if we just create the `apply` method with the default parameter for `boundaries`? That will break the binary compatibility, but not source compatibility. `<init>$default$2`, `apply$default$2` looks pretty bizarre IMO..

Reply:
Clever, but I don't love that this ends up being a breaking change with no compile time advice how to fix it.  I'd say we could publish a scalafix, but most users aren't running it, and lifting multipart creation into IO is a pretty big change.  It's naughty and impure, but it's probably not an area people are doing a lot of reasoning about.  I think we hang our heads in shame now and fix it for 1.0.

Reply:
FTR, we can make this package-private. It will remove static forwarders, which we usually deem as "ok".

Reply:
So the real question is still whether this impurity is so egregious that we should break apps for people who don't read the docs or release notes in a patch version.  I'm still leaning no, but I could be dissuaded.

Reply:
> Clever, but I don't love that this ends up being a breaking change with no compile time advice how to fix it.

My initial intention was to don't break compile time. Probably I'm missing something dully obvious, but providing that `apply` method:
```scala
@deprecated
def apply[F[_]](parts: Vector[Part[F]], boundary: Boundary = Boundary.unsafeCreate()) = new Multipart(parts, boundary)
```
it should work well for goals of preserving source compatibility (stable compilation).

Reply:
But yeah, that will steal the default case class constructor. And if someone used a fully specified constructor (simply put - didn't use a default parameter of the constructor), it will mean some inconvenience. Not so user-friendly indeed.

Reply:
Arman's technique is binary compatible, and with a deprecated unary apply and deprecated auxiliary constructor, we can make it source compatible.